src/extractor/extraction_containers.cpp

```diff
@@ -163,6 +199,7 @@ void ExtractionContainers::PrepareData(ScriptingEnvironment &scripting_environme
     PrepareNodes();
     WriteNodes(file_out);
     PrepareEdges(scripting_environment);
+    NodeVector().swap(all_nodes_list);
```

Comment:
The clear/shrink-to-fit pattern doesn't guarantee that the reserved memory is released.  The swap method is the recommended approach to ensure the memory reserve is released - see http://www.cplusplus.com/reference/vector/vector/clear/

I think a comment here would be good though, this is a c++ smell 😷